lib/grandpa/grandpa.go

```diff
@@ -962,7 +886,8 @@ func (s *Service) getPreVotedBlock() (Vote, error) {
 
 	// if there are multiple, find the one with the highest number and return it
 	highest := Vote{
-		Number: uint32(0),
+		Hash:   s.head.Hash(),
+		Number: uint32(s.head.Number),
 	}
```

Comment:
we don't need to set this because `blocks > 0` so the `highest` will definitely be set in the next for loop

Reply:
I would like to remove the `waitForFirstBlock` method as it is normal to gossip votes for the genesis block, that being said if we do not set the infos so when we call the method `getGrandpaGHOST` and the `len(blocks) > 0` the following problem happens:

```go
// blocks -> map[hash]uint32
blocks := { 0xABC: 0 }

// Without setting
// -------

highest := Vote {Number: uint32(0)}
for h, n := range blocks {
  // this condition will never happens as 0 > 0 is false 
  // and then the highest will have an empty hash
  if n > highest.Number {...}
}

// Setting latest finalized info
// -------

highest := Vote {Hash: s.head.Hash(), Number: uint32(s.head.Number)}
for h, n := range blocks {
  // this condition will never happens as 0 > 0 is false 
  // but we still have informations in the highest in a case where there is no highest blocks
  if n > highest.Number {...}
}
```

I think that we should set the highest as the latest finalized block (aka `s.head.Hash()` and `s.head.Number`) and then compare it with the incoming possible selected blocks, that way we avoid choosing a block that is before the latest finalized block as well. what do you think?